lsp-mode.el

```diff
@@ -5388,41 +5388,41 @@ returns the command to execute."
 (defun lsp--create-sentinel (workspace)
   "Create sentinel handler for WORKSPACE."
   (lambda (process exit-str)
-    (let* ((status (process-status process))
-           (folder->workspaces (lsp-session-folder->servers (lsp-session)))
-           (stderr (-> workspace lsp--workspace-proc process-name get-buffer)))
+    (unless (process-live-p process)
+      (let* ((status (process-status process))
+             (folder->workspaces (lsp-session-folder->servers (lsp-session)))
+             (stderr (-> workspace lsp--workspace-proc process-name get-buffer)))
 
-      (when (memq status '(exit signal))
         (lsp--warn "%s has exited (%s)"
                    (process-name (lsp--workspace-proc workspace))
-                   (string-trim-right exit-str)))
-
-      (with-lsp-workspace workspace
-        ;; clean workspace related data in each of the buffers
-        ;; in the workspace.
-        (--each (lsp--workspace-buffers workspace)
-          (when (buffer-live-p it)
-            (with-current-buffer it
-              (setq lsp--buffer-workspaces (delete workspace lsp--buffer-workspaces))
-              (lsp--uninitialize-workspace)
-              (lsp--spinner-stop)
-              (lsp--remove-cur-overlays))))
-
-        ;; cleanup session from references to the closed workspace.
-        (--each (hash-table-keys folder->workspaces)
-          (lsp--update-key folder->workspaces it (apply-partially 'delete workspace)))
-
-        ;; Kill standard error buffer only if the process exited normally.
-        ;; Leave it intact otherwise for debugging purposes.
-        (when (and (eq status 'exit) (zerop (process-exit-status process)) (buffer-live-p stderr))
-          (kill-buffer stderr)))
-
-      (run-hook-with-args 'lsp-after-uninitialized-hook workspace)
-
-      (if (eq (lsp--workspace-shutdown-action workspace) 'shutdown)
-          (lsp--info "Workspace %s shutdown." (lsp--workspace-print workspace))
-        (lsp--restart-if-needed workspace))
-      (lsp--cleanup-hanging-watches))))
+                   (string-trim-right exit-str))
+
+        (with-lsp-workspace workspace
+          ;; clean workspace related data in each of the buffers
+          ;; in the workspace.
+          (--each (lsp--workspace-buffers workspace)
+            (when (buffer-live-p it)
+              (with-current-buffer it
+                (setq lsp--buffer-workspaces (delete workspace lsp--buffer-workspaces))
+                (lsp--uninitialize-workspace)
+                (lsp--spinner-stop)
+                (lsp--remove-cur-overlays))))
+
+          ;; cleanup session from references to the closed workspace.
```

Comment:
We can capitalize comments since we are updating them.

Reply:
Done, but I think we do not follow this style guide.